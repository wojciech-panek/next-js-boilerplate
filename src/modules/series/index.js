export {
  SeriesTypes,
  SeriesActions,
} from './series.redux';

export * from './series.selectors';
