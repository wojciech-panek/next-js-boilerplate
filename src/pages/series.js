import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'next/router';
import { compose } from 'ramda';

import withIntl from '../shared/lib/withIntl';


export class SeriesPage extends PureComponent {
  static propTypes = {
    router: PropTypes.object.isRequired,
  };

  render() {
    return (
      <div>Series: {this.props.router.query.id}</div>
    );
  }
}

export default compose(
  withIntl,
  withRouter,
)(SeriesPage);
