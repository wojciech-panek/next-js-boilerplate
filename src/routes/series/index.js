export { default as Series } from './series.container';
