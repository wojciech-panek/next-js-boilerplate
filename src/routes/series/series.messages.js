/* eslint-disable max-len */
import { defineMessages } from 'react-intl';


export default defineMessages({
  pageTitle: {
    id: 'series.pageTitle',
    defaultMessage: 'Series',
  },
  season: {
    id: 'series.season',
    defaultMessage: 'Season',
  },
  episodes: {
    id: 'series.episodes',
    defaultMessage: 'Episodes',
  },
  back: {
    id: 'series.back',
    defaultMessage: 'Back',
  },
});
