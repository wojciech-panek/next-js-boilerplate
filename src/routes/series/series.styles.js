import styled from 'styled-components';


export const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  text-align: center;
`;

export const EpisodesContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  text-align: center;
  margin-bottom: 20px;
`;

