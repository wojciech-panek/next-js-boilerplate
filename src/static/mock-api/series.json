[
  {
    "id": 1,
    "title": "Dick in Ibiza",
    "season": "1"
  },
  {
    "id": 2,
    "title": "Dick in Ibiza",
    "season": "2"
  },
  {
    "id": 3,
    "title": "Liars",
    "season": "1"
  },
  {
    "id": 4,
    "title": "F*ck My Life",
    "season": "1"
  }
]
