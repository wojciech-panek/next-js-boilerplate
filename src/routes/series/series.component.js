import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import Head from 'next/head';

import { Loader } from '../../shared/components/loader';
import { Link } from '../../shared/components/link';
import { H2, H4 } from '../../shared/theme/typography';
import { Container, EpisodesContainer } from './series.styles';
import messages from './series.messages';



export class Series extends PureComponent {
  static propTypes = {
    data: PropTypes.object,
    router: PropTypes.object.isRequired,
  };

  getData = () => {
    if (this.props.data.size > 0 && this.props.data.get('id') === parseInt(this.props.router.query.id, 10)) {
      return this.props.data;
    }
    return null;
  };

  renderEpisode = (episode) => {
    return (
      <Link key={episode.get('id')} to={`/episodes/${episode.get('id')}`}>
        <a>[{episode.get('number')}] {episode.get('title')}</a>
      </Link>
    );
  };


  render() {
    const data = this.getData();

    if (!data) {
      return <Loader />;
    }

    return (
      <Container>
        <FormattedMessage {...messages.pageTitle}>
          {pageTitle => (
            <Head>
              <title key="title">{ pageTitle }</title>
            </Head>
          )}
        </FormattedMessage>

        <H2>{data.get('title')} [<FormattedMessage {...messages.season} />: {data.get('season')}]</H2>

        <H4><FormattedMessage {...messages.episodes} />:</H4>

        <EpisodesContainer>
          {data.get('episodes').toArray().map(this.renderEpisode)}
        </EpisodesContainer>

        <Link to="/">
          <a><FormattedMessage {...messages.back} /></a>
        </Link>
      </Container>
    );
  }
}
