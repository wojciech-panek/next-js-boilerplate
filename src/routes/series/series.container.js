import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { createStructuredSelector } from 'reselect';
import { compose } from 'ramda';
import { withRouter } from 'next/router';

import withIntl from '../../shared/lib/withIntl';
import { selectSeriesData } from '../../modules/series';
import { Series } from './series.component';


const mapStateToProps = createStructuredSelector({
  data: selectSeriesData,
});

export const mapDispatchToProps = (dispatch) => bindActionCreators({}, dispatch);

export default compose(
  withIntl,
  withRouter,
  connect(mapStateToProps, mapDispatchToProps),
)(Series);
