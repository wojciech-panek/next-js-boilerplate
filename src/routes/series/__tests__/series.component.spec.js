import React from 'react';
import { shallow } from 'enzyme';
import { fromJS } from 'immutable';

import { Series } from '../series.component';


describe('Series: Component', () => {
  const defaultProps = {
    router: { query: { id: 1 } },
    data: fromJS({
      id: 1,
      title: 'example series title',
      episodes: [
        { id: 1, number: 1, title: 'example episode title 1' },
        { id: 2, number: 2, title: 'example episode title 2' },
        { id: 3, number: 3, title: 'example episode title 3' },
      ],
    }),
  };

  const component = (props) => (
    <Series {...defaultProps} {...props} />
  );

  const render = (props = {}) => shallow(component(props));

  it('should render correctly when data is provided', () => {
    const wrapper = render();
    global.expect(wrapper).toMatchSnapshot();
  });

  it('should render <Loader /> when data is not provided', () => {
    const wrapper = render({ data: fromJS({}) });
    global.expect(wrapper).toMatchSnapshot();
  });

  it('should render <Loader /> when data does not match', () => {
    const wrapper = render({ router: { query: { id: 2 } } });
    global.expect(wrapper).toMatchSnapshot();
  });
});
