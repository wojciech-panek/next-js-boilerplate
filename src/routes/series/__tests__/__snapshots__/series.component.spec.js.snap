// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Series: Component should render <Loader /> when data does not match 1`] = `<InjectIntl(Loader) />`;

exports[`Series: Component should render <Loader /> when data is not provided 1`] = `<InjectIntl(Loader) />`;

exports[`Series: Component should render correctly when data is provided 1`] = `
<seriesstyles__Container>
  <FormattedMessage
    defaultMessage="Series"
    id="series.pageTitle"
    values={Object {}}
  >
    <Component />
  </FormattedMessage>
  <typography__H2>
    example series title
     [
    <FormattedMessage
      defaultMessage="Season"
      id="series.season"
      values={Object {}}
    />
    : 
    ]
  </typography__H2>
  <typography__H4>
    <FormattedMessage
      defaultMessage="Episodes"
      id="series.episodes"
      values={Object {}}
    />
    :
  </typography__H4>
  <seriesstyles__EpisodesContainer>
    <InjectIntl(withRouter(Connect(Link)))
      key="1"
      to="/episodes/1"
    >
      <a>
        [
        1
        ] 
        example episode title 1
      </a>
    </InjectIntl(withRouter(Connect(Link)))>
    <InjectIntl(withRouter(Connect(Link)))
      key="2"
      to="/episodes/2"
    >
      <a>
        [
        2
        ] 
        example episode title 2
      </a>
    </InjectIntl(withRouter(Connect(Link)))>
    <InjectIntl(withRouter(Connect(Link)))
      key="3"
      to="/episodes/3"
    >
      <a>
        [
        3
        ] 
        example episode title 3
      </a>
    </InjectIntl(withRouter(Connect(Link)))>
  </seriesstyles__EpisodesContainer>
  <InjectIntl(withRouter(Connect(Link)))
    to="/"
  >
    <a>
      <FormattedMessage
        defaultMessage="Back"
        id="series.back"
        values={Object {}}
      />
    </a>
  </InjectIntl(withRouter(Connect(Link)))>
</seriesstyles__Container>
`;
